OneCommunityGlobal/HGNRest
# Sav - Refactor Total Hours in Category Badge function

# Total hours badges auto assigning.
This bug detailed that the economics badge for 200Hrs was being assigned to users who did not have 200 hrs for that category. I was not able to find this specific bug but I did find some refactoring that could be implemented to make the function more readable and changed the badge condition so that any user without the prerequisite hours for the lowest tier badge would not be tested. 

Fixes #9 Badge Component Bug Priority Medium

## Mainly changes explained:
Changed the conditional for checking the user hours against the badge hours.
Refactored the badgesOfType array to be more readable
…

## How to test:
- Check into the current branch
- Do `npm install` and `...` to run this PR locally
- Log in as an admin user
- Go to Profile -> Volunteering times 
- Change total hours in categories to make sure badges are being assigned properly

## Screenshots or videos of changes:
![Screenshot from 2023-03-23 16-30-16](https://user-images.githubusercontent.com/98275089/227349787-cdbc960c-1ad0-4b73-91ae-8c61e15a6313.png)
![Screenshot from 2023-03-23 16-29-59](https://user-images.githubusercontent.com/98275089/227349820-72f1541a-de90-4f2a-a4c2-d9b5efb22cb2.png)

## Note:
With badge updates, there is a bit of a wait while things update on the backend, allow usually a minimum of 5 minutes for things to change.
